src/main/java/com/afetyardim/afetyardim/service/istanbul/IstanbulSitesParser.java

```diff
@@ -0,0 +1,344 @@
+package com.afetyardim.afetyardim.service.istanbul;
+
+import com.afetyardim.afetyardim.model.*;
+import com.afetyardim.afetyardim.service.SiteService;
+import com.afetyardim.afetyardim.service.common.SpreadSheetUtils;
+import com.afetyardim.afetyardim.service.common.SiteUtils;
+import com.google.api.services.sheets.v4.model.Color;
+import com.google.api.services.sheets.v4.model.RowData;
+import com.google.api.services.sheets.v4.model.Spreadsheet;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.*;
+
+import static com.afetyardim.afetyardim.service.common.SiteUtils.compareFloats;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class IstanbulSitesParser {
+
+    private final SpreadSheetUtils spreadSheetUtils;
+    private final SiteService siteService;
+
+    private final static String ISTANBUL_SPREADSHEET_ID = "1B0epkFl-4dF4FINjTSONWbHivOu702RBIWYnOM1gWkg";
+    private final static String ISTANBUL_SPREADSHEET_RANGE = "A1:G300";
+    private final static String CITY_NAME = "İstanbul";
+
+    public void parseSpreadsheet() throws IOException {
+        log.info("Start Istanbul spread sheet parsing");
+
+        Spreadsheet spreadsheet = spreadSheetUtils.getSpreadSheet(ISTANBUL_SPREADSHEET_ID, ISTANBUL_SPREADSHEET_RANGE);
+        List<RowData> rows = spreadsheet.getSheets().get(0).getData().get(0).getRowData();
+        //Remove header row
+        rows.remove(0);
+        Collection<Site> istanbulSites = siteService.getSites(Optional.of(CITY_NAME), Optional.empty());
+        List<Site> newSites = new ArrayList<>();
+        int updatedSiteCount = 0;
+
+        for (int i = 0; i < rows.size(); i++) {
+            RowData row = rows.get(i);
+            try {
+
+                String siteName = (String) row.getValues().get(2).get("formattedValue");
+                if (siteName == null) {
+                    log.info("Could not get siteName, returning");
+                    return;
+                }
+
+                String district;
+                try{
+                    district = (String) row.getValues().get(0).get("formattedValue");
+                }catch(Exception ex){
+                    district = null;
+                }
+
+                if (siteName == null || district == null) {
```

Comment:
siteName null check is redundant, please remove it

Reply:
Fixed